Assets/Scripts/Environment/ObstacleMovement.cs

```diff
@@ -4,8 +4,7 @@
 
 public class ObstacleMovement : MonoBehaviour
 {
-    [SerializeField]
-    private float movementSpeed = 1f;
```

Comment:
There is no need to expose the field by making it public, better add a property to expose this field.